core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -164,11 +167,25 @@ long memorySize() {
 
         updateIndexAndNormalize(engaged, newChunks);
 
-        for (Chunk chunk : engaged
-        ) {
+        for (Chunk chunk : engaged) {
+            ConcurrentSkipListSet<Iter> viewingIterators = chunk.getSignedIterators();
             chunk.release();
-        }
+            viewingIterators.forEach(iterator -> {
+                if (iterator.getNextKeyIndex() != Chunk.NONE) {
```

Comment:
call getNextKeyIndex() once and compare result twice
is it safe to call it twice? can the index change? if not add a comment explaining exactly why it has not changed?

Reply:
Ill make this code more clear. I can call getNext only once and its ok.